### PR TITLE
Feat: Combat attacks

### PR DESCRIPTION
**Pull request by PedroChaparro**
### Includes

- [x] Automated gym protectors attack
- [x] Receive user attacks 
- [x] Listen for the user to dodge attacks
- [x] 10% probability for the gym Loomie to dodge attacks 
- [x] Timeout to prevent user to spam attacks
- [x] Finish the combat  and close the WS connection when the user wins / looses.  
- [x] Base function to handle when a gym Loomie is weakened in combat.  

--- 
**NOTE:** You must set the new environment variables. Refer to the `.env.example` file for more information.

### CODE REVIEW

**Review by andres123dbh — APPROVED** [review]

**Comment by PedroChaparro** [comment]
I would like to wait for @Woynert approval.

**Review by SilviaPabon — APPROVED** [review]

**Pull request merged**
